.github/workflows/smoke-test-imperative-compare.yml

```diff
@@ -0,0 +1,124 @@
+name: Imperative Smoke Tests Compare
+concurrency:
+  # Group PRs by head_ref, push to main branch by commit id, and others branch by ref.
+  group: ${{ github.workflow }}-${{ github.head_ref || (github.ref == 'refs/heads/main' && github.sha) || github.ref }}
+  cancel-in-progress: true
+on: [push, pull_request]
+env:
+  JHI_SAMPLES: ${{ github.workspace }}/generator-jhipster-quarkus/test-integration/samples/
+jobs:
+  applications:
+    name: ${{ matrix.app }}
+    runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]')"
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        node_version: [14.16.0]
+        os: [ubuntu-latest]
+        app:
+          - imperative-ngx-jwt-maven-mysql-caffeine
+          - imperative-ngx-jwt-maven-mongodb-caffeine
+          - imperative-ngx-jwt-gradle-mongodb-redis
+          - imperative-ngx-jwt-maven-psql-nocache
+          - imperative-react-jwt-gradle-mysql-caffeine
+          - imperative-ngx-jwt-maven-mysql-redis
+          - imperative-sql-jwt-maven-no-db
+          - imperative-ngx-oauth2-maven-mysql-caffeine
+        include:
+          - app: imperative-ngx-jwt-maven-mysql-caffeine
+          - app: imperative-ngx-jwt-maven-mongodb-caffeine
+          - app: imperative-ngx-jwt-maven-psql-nocache
+          - app: imperative-react-jwt-gradle-mysql-caffeine
+          - app: imperative-ngx-jwt-maven-mysql-redis
+          - app: imperative-sql-jwt-maven-no-db
+          - app: imperative-ngx-oauth2-maven-mysql-caffeine
+        exclude:
+          - app: imperative-ngx-jwt-gradle-mongodb-redis
+    steps:
+      - name: 'SETUP: Checkout quarkus blueprint'
+        uses: actions/checkout@v2
+        with:
+          path: generator-jhipster-quarkus
+          fetch-depth: 5
+      - name: git history
+        run: git --no-pager log -n 10 --graph --pretty='%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue) <%an>%Creset' --abbrev-commit
+        working-directory: ./generator-jhipster-quarkus
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v2
+        with:
+          repository: 'jhipster/generator-jhipster'
```

Comment:
It seems we are relying on the main branch of `JHipster`. I suggest to rely on the current supported JHipster version i.e. `v7.9.3` to avoid any surprises.

Reply:
Yes. It should be version specific.